simphony_osp/session/session.py

```diff
@@ -78,7 +62,12 @@ def locked(self) -> bool:
         manager and leaving the context. Useful for setting it as the
         default environment when it is not intended to close it afterwards.
         """
-        return self._lock > 0
+        return (self._lock + bool(self._user_lock)) > 0
+
+    @locked.setter
+    def locked(self, value: bool):
+        """Lock or unlock an environment."""
```

Comment:
You have a few references to this docstring
> See the docstring of `locked` for an explanation of what locking an environment means

But I feel like this does not explain much :)

Reply:
You are reading the docstring of the _setter_ of the property. Check the [docstring of the _getter_](https://github.com/simphony/osp-core/blob/1c7692944c085b4e5ae93d256a9755804d390b58/simphony_osp/session/session.py#L58) a few lines above. I did not foresee that this misunderstanding could occur :sweat_smile:. 

In any case you have a better explanation below. If you think it is a good example, it can be included in the docstring itself.

**No locking**

```python
session = MyWrapper()  # -> the session is now open
with session:
    pass
# -> the session is now closed
```

**Locking feature**

``````python
session = MyWrapper()  # -> the session is now open
session.locked = True
with session:
    pass
# -> the session is still open
```

Of course the use case is setting the default session temporarily to create ontology individuals inside it, without having to close it at the end (for example to do stuff in another session, without having to nest another with statement).

Reply:
My bad, I was reading the diff between the branches and did not notice it was the setter. I think it is alright then :)